lib/model/model.go

```diff
@@ -2882,16 +2882,30 @@ func (m *model) Availability(folder string, file protocol.FileInfo, block protoc
 	}
 	defer snap.Release()
 
-	return m.availabilityInSnapshotRLocked(cfg, snap, file, block), nil
+	if devices := m.availabilityInSnapshot(cfg, snap, file); len(devices) > 0 {
+		return devices, nil
+	}
+
+	return m.availabilityFromTemporaryRLocked(cfg, file, block), nil
 }
 
-func (m *model) availabilityInSnapshot(cfg config.FolderConfiguration, snap *db.Snapshot, file protocol.FileInfo, block protocol.BlockInfo) []Availability {
+func (m *model) availabilityInSnapshot(cfg config.FolderConfiguration, snap *db.Snapshot, file protocol.FileInfo) []Availability {
 	m.mut.RLock()
 	defer m.mut.RUnlock()
-	return m.availabilityInSnapshotRLocked(cfg, snap, file, block)
+	return m.availabilityInSnapshotRLocked(cfg, snap, file)
+}
+
+func (m *model) availabilityFromTemporaryRLocked(cfg config.FolderConfiguration, file protocol.FileInfo, block protocol.BlockInfo) []Availability {
+	var availabilities []Availability
+	for _, device := range cfg.Devices {
+		if m.deviceDownloads[device.DeviceID].Has(cfg.ID, file.Name, file.Version, int(block.Offset/int64(file.BlockSize()))) {
+			availabilities = append(availabilities, Availability{ID: device.DeviceID, FromTemporary: true})
+		}
+	}
+	return availabilities
```

Comment:
optinal suggestion: If you move this function below `availabilityInSnapshotRLocked` the diff might be smaller and also the order matches the reading order in `Availability`.